pkg/chunk/storage_client.go

```diff
@@ -30,6 +31,9 @@ type ObjectAndIndexClient interface {
 // WriteBatch represents a batch of writes.
 type WriteBatch interface {
 	Add(tableName, hashValue string, rangeValue []byte, value []byte)
+	Increment(tableName, hashValue string, rangeValue []byte, value int64)
```

Comment:
Starting a separate comment thread to talk about the usage of the IndexClient separately from CacheGen.  IMO this is a misuse of the IndexClient. It adds a number of functions to the WriteBatch interface that are entirely unrelated to the Index. It reuses parameters with the names `hashValue` and `rangeValue` which don't any real relation to what we are trying to do when storing Deletion Plans.

A separate client for deletes is a much cleaner way to do this. We don't need a `WriteBatch` for storing deletion plans or updating their status. From my perspective, the only thing we would need is a few functions, specifically:`CreatePlan`, `ListPlans` and `UpdatePlan`. This would make it obvious what the client is doing.

It also means the `ObjectClient` and `IndexClient` only have to implement one function respectively to support deletes. If the `DeleteClient`/`DeletePlanClient` isn't supported in a particular backend, that's fine because they could always use another backend they do support concurrently to serve as their delete client. I am also suggesting this because I think the range of clients that could support deletes is much broader that ones that support the index. We could be implementing this on top of S3/GCS, or a KVStore for example.